rosetta-source/src/main/rosetta/product-qualification-func.rosetta

```diff
@@ -27,168 +57,154 @@ func Qualify_AssetClass_InterestRate: <"Qualifies a product as having the Asset
-                (economicTerms -> payout -> interestRatePayout, economicTerms -> payout -> performancePayout) only exists
-                    // Performance Payout + Fixed Price Payout (Dividend Swap)
-                    or (economicTerms -> payout -> performancePayout, economicTerms -> payout -> fixedPricePayout) only exists
-                        // Performance Payout only (Variance, Volatility and Correlation Swap)
-                    or economicTerms -> payout -> performancePayout only exists))
+        (economicTerms -> payout -> performancePayout -> underlier -> Observable
+                extract Qualify_UnderlierObservable_Equity
+                then all = True
+            and ( // Interest Rate Payout + Performance Payout (Price Return Swap, Total Return Swap)
+            (economicTerms -> payout -> interestRatePayout, economicTerms -> payout -> performancePayout) only exists
+                // Performance Payout + Fixed Price Payout (Dividend Swap)
+                or (economicTerms -> payout -> performancePayout, economicTerms -> payout -> fixedPricePayout) only exists
+                    // Performance Payout only (Variance, Volatility and Correlation Swap)
+                or economicTerms -> payout -> performancePayout only exists))
             or (economicTerms -> payout -> optionPayout only exists
-                and if optionUnderlier exists
-                    then (Qualify_UnderlierProduct_Equity(optionUnderlier) = True
-                            or optionUnderlier -> index ->> assetClass = AssetClassEnum -> Equity
-                            or optionUnderlier -> security -> productTaxonomy -> primaryAssetClass any = AssetClassEnum -> Equity)
-                    else False)
-            or (economicTerms -> payout -> forwardPayout only exists
-                and if forwardUnderlier exists
-                    then (Qualify_UnderlierProduct_Equity(forwardUnderlier)
-                            or forwardUnderlier -> index ->> assetClass = AssetClassEnum -> Equity
-                            or forwardUnderlier -> security -> productTaxonomy -> primaryAssetClass any = AssetClassEnum -> Equity)
-                    else False)
+                and (ObservableQualification(optionUnderlier -> Observable, SecurityTypeEnum -> Equity, AssetClassEnum -> Equity) 
+                    or if optionUnderlier -> Product exists
+                        then Qualify_AssetClass_Equity(optionUnderlier -> Product ->> economicTerms) = True
+                        else False
+                ))
+            or (economicTerms -> payout -> settlementPayout only exists
+                and (UnderlierQualification(
+						settlementUnderlier, empty, AssetClassEnum -> Equity) 
+                    or if settlementUnderlier -> Product ->> economicTerms exists
+                        then Qualify_AssetClass_Equity(
+									settlementUnderlier -> Product ->> economicTerms) = True
```

Comment:
Same comment, is that qualification based on the settlement payout underlier's economic terms meant to cover the ETD case?